kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ActiveAddFilesIterator.java

```diff
@@ -233,9 +233,13 @@ private void prepareNext() {
       }
     }
 
+    ColumnarBatch scanAddFiles = addRemoveColumnarBatch;
     // Step 3: Drop the RemoveFile column and use the selection vector to build a new
     //         FilteredColumnarBatch
-    ColumnarBatch scanAddFiles = addRemoveColumnarBatch.withDeletedColumnAt(1);
+    // For Parquet files, we would only have read the adds, not the removes, hence the check.
+    if (scanAddFiles.getSchema().length() > 1) {
```

Comment:
This could also use `isFromCheckpoint`, which I maybe prefer in case in the future there's ever additional columns added or anything?

Reply:
Open to opinions

Reply:
I think that is cleaner. Thanks!